### PR TITLE
Sk-24 Finished Logic for LinkedIn Scraping with Multiple Education and Experience

### PR DESCRIPTION
**Pull request by Joyliu290**

### CODE REVIEW

File: profiles/models.py [hunk 1]
```diff
@@ -46,14 +46,13 @@ def education_to_vector(self, profile_simulation_date):
...
 
-
 class Experience(models.Model):
     profile = models.ForeignKey(Profile, on_delete=models.CASCADE)
     company = models.CharField(max_length=128, default='')
     title = models.CharField(max_length=32, default='')
     description = models.CharField(max_length=128, default='')
-    start_date = models.DateField(default='')
-    end_date = models.DateField(default='')
+    start_date = models.DateField(null = True, blank = True)
```
Comment (wangwalton) [inline profiles/models.py]: Spacing consistency with arguments, use `null=true`

File: profiles/linkedin.py [hunk 2]
```diff
@@ -11,43 +11,60 @@ def __init__(self):
...
+        :param company: Company name we're searching.
+        :param role: Role we're searching for.
+        :param num_profiles: Required number of profiles to fetch and store.
+        :return: List of Query objects of the profiles stored in the Profiles database that corresponds to the company and title.
+        """
         # first check in the database to see there are enough num_profiles
         # then call get_linkedin_urls_google_search, and then pass those urls to the scraper
         # and then store in the db
         profiles = []
         exclude_urls = []
```
Comment (wangwalton) [inline profiles/linkedin.py]: No need for this declaration, and set below to `profiles = [...]`

File: profiles/linkedin.py [hunk 3]
```diff
@@ -11,43 +11,60 @@ def __init__(self):
...
-                        experience__title__contains=title)]
-        exclude_urls += [p.profile_url for p in Profile.objects.filter(experience__company__contains=company, 
-                        experience__title__contains=title)]
+        profiles += [p for p in Profile.objects.filter(experience__company__icontains=company, 
+                        experience__title__icontains=title)]
+        exclude_urls += [p.profile_url for p in profiles]
 
-        linkedin_scraper= LinkedInScraper()
-        linkedin_scraper.login()
+        self.scraper.login()
```
Comment (wangwalton) [inline profiles/linkedin.py]: Move this to init actually, logging every time this function is called may seem suspicious

File: profiles/scraper.py [hunk 4]
```diff
@@ -194,117 +202,139 @@ def login(self):
...
+            elif num_profiles_saved_to_db>=num_profiles_needed:
+                return num_profiles_saved_to_db, exclude_urls
+
             time.sleep(2)
 
-        return num_profiles_saved_to_db
+        return num_profiles_saved_to_db, exclude_urls
 
-    def get_profiles(self, company, role, num_profiles=10, exclude_urls=[]): # get number of profiles as specified in the arg and store them in database
+    # get number of profiles as specified in the arg and store them in database
```
Comment (wangwalton) [inline profiles/scraper.py]: rm

File: profiles/scraper.py [hunk 5]
```diff
@@ -194,117 +202,139 @@ def login(self):
...
+
+        while num_profiles_saved_to_db < num_profiles and iteration < 5:
+            num_profiles_needed = num_profiles-num_profiles_saved_to_db
+            profile_links = self.collect_basic_profile(company, role, num_profiles_needed, exclude_urls)
+            num_new_profiles_saved_to_db, new_exclude_urls = self.store_profiles(profile_links, company, role, num_profiles_needed)
+            exclude_urls += new_exclude_urls
+            num_profiles_saved_to_db += num_new_profiles_saved_to_db
+            iteration+=1
 
     # returning a boolean to indicate if we have stored the profile into DB
```
Comment (wangwalton) [inline profiles/scraper.py]: rm

File: profiles/models.py (hunk 1 above)
Comment (Joyliu290) [reply profiles/models.py]: done

File: profiles/linkedin.py (hunk 2 above)
Comment (Joyliu290) [reply profiles/linkedin.py]: done

File: profiles/linkedin.py (hunk 3 above)
Comment (Joyliu290) [reply profiles/linkedin.py]: done

File: profiles/scraper.py (hunk 4 above)
Comment (Joyliu290) [reply profiles/scraper.py]: done

File: profiles/scraper.py (hunk 5 above)
Comment (Joyliu290) [reply profiles/scraper.py]: done

**Pull request merged**